### PR TITLE
ZK-3811: PathELResolver.getValue cause a NPE

### PR DESCRIPTION
**Pull request by klyvechen**

### CODE REVIEW

**Comment by klyvechen** [comment]
reopen

File: zkbind/src/org/zkoss/bind/xel/zel/PathELResolver.java [hunk 1]
```diff
@@ -51,8 +51,8 @@ public Object getValue(ELContext ctx, Object base, Object property)
 			numOfKids = (Integer) ctx.getContext(AstIdentifier.class); //Number of siblings of AstIdentifier
 			path = new Path();
 		} else {
-			numOfKids = _numOfKids.removeFirst();
-			path = _paths.removeFirst();
+			numOfKids = (!_numOfKids.isEmpty()) ? _numOfKids.removeFirst() : null;
+			path = (!_paths.isEmpty()) ? _paths.removeFirst() : new Path();
```
[the remark this answers is not in the document]
Comment (klyvechen) [reply zkbind/src/org/zkoss/bind/xel/zel/PathELResolver.java]: ok

File: zkbind/src/org/zkoss/bind/xel/zel/BindELResolver.java [hunk 2]
```diff
@@ -324,11 +324,11 @@ protected void tieValue(ELContext elCtx, Object base, Object property, Object va
 						final boolean prompt = bctx != null && bctx.getCommandName() == null;
 						if (prompt) {
 							//FormBinding shall not check @DependsOn() for dependent nodes
-							String basePath = path.getTrackBasePath();
+							String basePath = (path != null) ? path.getTrackBasePath() : null;
 							boolean skipAddingDependsOn = (basePath == null || basePath.length() == 0); //Should ignore empty base
 							if (!skipAddingDependsOn && (!(binding instanceof LoadFormBindingImpl)
 									|| ((LoadFormBindingImpl) binding).getSeriesLength() <= path.size())) {
-								BindELContext.addDependsOnTrackings(m, basePath, path.getTrackFieldsList(), binding,
+								BindELContext.addDependsOnTrackings(m, basePath, (path != null) ? path.getTrackFieldsList() : null, binding,
```
[the remark this answers is not in the document]
Comment (klyvechen) [reply zkbind/src/org/zkoss/bind/xel/zel/BindELResolver.java]: ok

File: zktest/src/archive/test2/config.properties [hunk 3]
```diff
@@ -2901,6 +2901,8 @@ F85-ZK-3824.zul=A,E,AuWriters,AuResponse,CONTENT_TYPE
 ##ztl##F85-ZK-3704.zul=A,E,NumberInputElement,FormatWidget,Intl
 ##zats##F85-ZK-3846.zul=A,E,Datebox,SimpleDateConstraint,format,separator
 F85-ZK-3852.zul=A,E,Chosenbox,inplace
+##ztl##B85-ZK-3945.zul=A,E,zindex,popup,window
+##zats##B85-ZK-3811.zul=A,E,PathELResolver,BindELResolver,getValue,tieValue
```
Comment (scribetw) [inline zktest/src/archive/test2/config.properties]: 1. wrong place
2. B85-ZK-3945.zul shouldn't be here

Comment (klyvechen) [reply zktest/src/archive/test2/config.properties]: ok sorry

**Comment by wenninghsu** [comment]
@zkadmin rebase

**Comment by zkadmin** [comment]
I just pushed up the changes, enjoy!

**Pull request merged**